packages/picasso.js/src/core/component/tween.js

```diff
@@ -76,8 +76,14 @@ export default function tween({ old, current }, { renderer }, config) {
         tweens: entered.ips,
         nodes: [...updated.nodes],
       });
-      if (config.isMainComponent && toBeUpdated.length === 0) {
-        shouldRemoveUpdatingStage = true;
+      if (config.isMainComponent) {
+        const filterFn = config.isMainComponent?.filterFn;
+        const nUpdatingNodes = filterFn ? toBeUpdated.filter(filterFn).length : toBeUpdated.length;
+        if (shouldRemoveUpdatingStage === undefined) {
```

Comment:
I do think it's a bit odd to have a variable that can be both a boolean and undefined. Also the name of it hints that it's a boolean which makes it even more odd.

Reply:
It is a boolean. When I assign `undefined` to it, the purpose is to say: its value is not defined yet (we don't know if we should remove updating stage at that moment). What would you do otherwise in this situation?

Reply:
Thanks for the suggestion. I have updated the code. Please take a look when you have time.